### ISSUE TITLE
Colorize keywords, arguments, variables, etc. in the editor

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#145]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110) by rover.zh... on 26 Feb 2009*

I suggest that in the editor,the keyword,arguments and variables can be set
with different color and format.If so ,it will be easy to edit the testcases~~

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c1) by @yanne on 13 Mar 2009*


I like this idea. However there are many basic features that are more important than
this, so this won't be included in the 1.0 release plan.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c2) by @pekkaklarck on 9 Jun 2009*


<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=153" title="colorize keywords"> Issue 153 </a> has been merged into this issue.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c4) by bryan.oakley on 18 Aug 2009*


For what it's worth, Orbitz has a plugin that supports this feature. We won't release
it until <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=203" title="Publish an event when a cell value changes in a grid"> issue #203 </a> is implemented because we had to access some private data
structures, making it somewhat fragile. Once <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=203" title="Publish an event when a cell value changes in a grid"> issue #203 </a> is implemented we'll be glad
to polish it up a little and submit it either as a downloadable package or as part of
the core.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c5) by @pekkaklarck on 19 Aug 2009*


Awesome! <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=203" title="Publish an event when a cell value changes in a grid"> Issue 203 </a> is targeted to the next release (0.18). Is it possible to include
this one too?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c6) by bryan.oakley on 19 Aug 2009*


Possibly. It needs more testing but, being a plugin, if it causes problems people can
just turn it off.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c7) by @pekkaklarck on 19 Aug 2009*


Let's try to get this in to 0.18 too then. No worries if we need to drop it, though.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c8) by @pekkaklarck on 15 Sep 2009*


Bryan, have you had time to look at this? he issue this one was blocked on has
already been fixed. We'll try to create 0.18 release on Wednesday and need to know
should we wait for this issue or not. There's definitely enough content for 0.18
already so dropping this and including it into 0.19 or later is fine too.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c9) by bryan.oakley on 15 Sep 2009*


I haven't had time to fully test it, but <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=203" title="Publish an event when a cell value changes in a grid"> issue 203 </a> doesn't looked fixed to me. Maybe
I'm doing something wrong but I don't get the cell change publication when I
interactively edit a cell. I do get the notification when a test is first loaded up,
but then nothing.

Since RIDE now depends on a newer version of wxPython than I can install on my box
(for crazy bureaucratic reasons I hope to eventually resolve) I don't have time to
debug any further.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c10) by bryan.oakley on 16 Sep 2009*


I have a plugin ready that works with 0.18 but at the last minute I discovered an
edge case and I won't have time to work on it today or tomorrow. So probably better
to wait for 0.19 unless 0.18 is going to be delayed by a couple of days

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c11) by oakley on 18 Sep 2009*


I've attached a simple plugin to add color to a cell when it changes. You may add it
to the RIDE distribution if you want. I've tested it with the latest changes as of
the time I write this. It's not a perfect solution but hopefully it will inspire
someone to make some improvements. 

A couple of known weaknesses: 

There's a small performance hit when loading a test case. I've tested it with a test
that has 1000 rows and there's a very perceptible delay when loading the data, but
there's a delay even without the colorizing effect so I'm not too worried about that.
For normal test cases I think the performance is acceptable. If performance becomes
an issue this should probably be done within the kweditor rather than as a plugin.

Also, if you activate or deactivate the plugin it doesn't automatically add or remove
color from an already loaded test case.  You have to manually cause the test to be
relaoded. 

Finally, if you don't like the colors you have to hand-edit the config file.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c12) by jrvilda on 18 Sep 2009*


Where is should be placed?
Do I need to modify the pluginmanager for add?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c13) by bryan.oakley on 18 Sep 2009*


I do not know where it should be placed, which is why I attached it here :-). I think
we need some sort of contribution section in the distribution. That is, unless you
want to make this part of the core, in which case it would go in src/robotide/application

If you personally want to try it out, just drop it in ~/.robotframework/ride/plugins
then activate it via the plugin manager

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c16) by jrvilda on 18 Sep 2009*


	
to prove it is necessary to modify the script pluginmanager.py and modify the name of
the file for kwcolorizer and place it in src \ robotide \ application


kwcolorizer import from Colorizer

CORE_PLUGINS = [ReleaseNotesPlugin, PluginManagerPlugin, RecentFilesPlugin,
ProgramadorPlugin, Colorizer
                 PreviewPlugin]

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c17) by jrvilda on 18 Sep 2009*


IN v0.18
Error on line 19
import Plugin from robotide.application
possible solution for v0.18
from plugin import Plugin

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c18) by bryan.oakley on 18 Sep 2009*


Hmmm. Looks like my testing was flawed -- I had both the old and new versions of RIDE
on my path. Sorry about the confusion.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c19) by @pekkaklarck on 18 Sep 2009*


Janne already committed this plugin to the core and also fixed the problems mentioned
above. I really like it! Thanks a lot Bryan, this is an awesome enhancement!

Should the plugin also register itself under Tools menu somehow? I'd like as many as
possible to start using this plugin because it's so cool. Or should we just enable it
by default and let those who encounter possible problems let disable it from the
plugin manager?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c20) by bryan.oakley on 18 Sep 2009*


I vote for the final option -- just enable it by default and let those who encounter
problems disable it from the plugin manager. There's no need to put it on the tools menu.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=110#c21) by @yanne on 21 Sep 2009*


Colorizer is now enabled by default and can be disabled via manage plugins.
